resource_customizations/argoproj.io/Rollout/actions/promote/action.lua

```diff
@@ -0,0 +1,54 @@
+function getCurrentCanaryStep(obj)
+
+    if obj.spec.strategy.canary == nil or table.getn(obj.spec.strategy.canary.steps) == 0 then
+        return nil
+    end
+
+	local currentStepIndex = 0
+
+	if obj.status.currentStepIndex ~= nil then
+		currentStepIndex = obj.status.currentStepIndex
+    end
+
+	return currentStepIndex
+
+end
+
+if obj.spec.paused ~= nil and obj.spec.paused then
+    obj.spec.paused = false
```

Comment:
Can you give more insight on when paused would be true and why do we need to handle it in promote?